source/features/conversation-timeline-filter.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import {observe} from 'selector-observer';
+import * as pageDetect from 'github-url-detection';
+import {CheckIcon, GearIcon, XIcon} from '@primer/octicons-react';
+
+import features from '.';
+
+enum FilterSettings {
+	ShowAll = 1,
+	ShowOnlyComments = 2,
+	ShowOnlyUnresolvedComments = 3,
+	ShowOnlyUnresolvedReviews = 4
+}
+
+let currentSettings: FilterSettings = FilterSettings.ShowAll;
+
+const showFilterName = 'rgh-show-filter';
+const menuItemCheckbox = 'rgh-filter-menu-item-checkbox';
+
+// Every element on the timeline that is recognizable by this feature will be marked with tis class.
+const timelineFiltersSelectorId = 'timeline-filters';
+
+const summaries = {
+	[FilterSettings.ShowAll]: '',
+	[FilterSettings.ShowOnlyComments]: 'Show only comments',
+	[FilterSettings.ShowOnlyUnresolvedComments]: 'Show only unresolved reviews',
+	[FilterSettings.ShowOnlyUnresolvedReviews]: 'Show unresolved comments'
+};
+
+function regenerateFilterSummary(): void {
+	select(`#${timelineFiltersSelectorId}`)!.textContent = summaries[currentSettings];
+}
+
+async function saveSettings(filterSettings: FilterSettings, test: string): Promise<any> {
+	currentSettings = filterSettings;
+	regenerateFilterSummary();
+	reapplySettings();
+
+	select.all('.' + menuItemCheckbox)
+		.forEach(element => {
+			element.style.display = 'none';
```

Comment:
Use the hidden property everywhere instead of this 
```suggestion
			element.hidden = true
```